homeassistant/components/benq_projector/switch.py

```diff
@@ -0,0 +1,199 @@
+"""
+Obtain and set state of a Benq projector over telnet-to-serial gateway.
+
+For more details about this component, please refer to the documentation
+at https://home-assistant.io/components/switch.benq_projector/
+"""
+import logging
+import threading
+
+import voluptuous as vol
+
+from homeassistant.components.switch import (SwitchDevice, PLATFORM_SCHEMA)
+from homeassistant.const import (
+    STATE_ON, STATE_OFF, STATE_UNKNOWN, CONF_NAME, CONF_HOST)
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = 'Benq Projector'
+DEFAULT_TIMEOUT = None
+DEFAULT_WRITE_TIMEOUT = None
+
+LAMP_MODE = 'Lamp Mode'
+ICON = 'mdi:projector'
+INPUT_SOURCE = 'Input Source'
+POWER = 'Power'
+LAMP_HOURS = 'Lamp Hours'
+MODEL = 'Model'
+
+# Commands known to the projector
+CMD_DICT = {
+    POWER: '*pow=?#',
+    LAMP_HOURS: '*ltim=?#',
+    INPUT_SOURCE: '*sour=?#',
+    LAMP_MODE: '*lampm=?#',
+    MODEL: '*modelname=?#',
+    STATE_ON: '*pow=on#',
+    STATE_OFF: '*pow=off#',
+}
+
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+})
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Connect with telnet host and return Benq Projector."""
+    host = config.get(CONF_HOST)
+    name = config.get(CONF_NAME)
+
+    add_entities([BenqSwitch(host, name)], True)
+
+
+class BenqSwitch(SwitchDevice):
+    """Represents an Benq Projector as a switch."""
```

Comment:
Docstrings shouldn't be in imperative mood.

Reply:
changed, better that way?